### PR TITLE
Lab2: Implementing Bubble and Insertion sort

### PR DESCRIPTION
**Pull request by sksandani**
Fixes #42
Code to display array in sorted order using bubble sort

#### Changes proposed in this PR:

- Implementing the bubble sort
- Implementing the insertion sort
- Test case to cover 100% code coverage

### CODE REVIEW

File: 02_bubble/sksandani/main.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
```
Comment (golangcibot) [inline 02_bubble/sksandani/main.go:9]: `out` is unused (from `varcheck`)

Comment (sksandani) [reply 02_bubble/sksandani/main.go:9]: Removed unused imports

File: 02_bubble/sksandani/main.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+func main() {
+	fmt.Fprintln(out, "Bubble: ", bubble([]int{3, 2, 1, 5}), "Insertion: ", insertion([]int{3, 2, 1, 5}))
```
Comment (juliaogris) [inline 02_bubble/sksandani/main.go]: Not the requested input/output

File: 02_bubble/sksandani/main.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+func bubble(s []int) []int {
+	n := len(s)
+	for j := n; j > 0; j-- {
+		for i := 1; i <= n-1; i++ {
+			if s[i-1] > s[i] {
```
Comment (juliaogris) [inline 02_bubble/sksandani/main.go]: Modifies the input, please make a copy.

File: 02_bubble/sksandani/main.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	}
+	return s
+}
+
+func insertion(s []int) []int {
+	n := len(s)
+	for i := 1; i < n; i++ {
+		for j := i; j > 0; j-- {
+			if s[j-1] > s[j] {
+				s[j], s[j-1] = s[j-1], s[j]
```
Comment (juliaogris) [inline 02_bubble/sksandani/main.go]: Modifies the input, please make a copy first.

File: 02_bubble/sksandani/main_test.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+func TestBubbleSort(t *testing.T) {
+	// Given
+	r := require.New(t)
+
+	// When
+	i := bubble([]int{3, 2, 1, 5}) // Then
+	expected := []int{1, 2, 3, 5}
+	actual := i
+	r.Equalf(expected, actual, "Unexpected output in main()")
```
Comment (juliaogris) [inline 02_bubble/sksandani/main_test.go]: Update "failure" message, below too.

**Review by juliaogris — COMMENTED** [review]
Please edit your commits to follow [recommendations](https://chris.beams.io/posts/git-commit/)
    -   Capitalize the subject line
    -   Use the imperative mood in the subject line
    -   Fold fixup commits into previous ones with `git rebase -i HASH`

File: 02_bubble/sksandani/main.go (hunk 2 above)
Comment (sksandani) [reply 02_bubble/sksandani/main.go]: Refactored the code print expected output

File: 02_bubble/sksandani/main.go (hunk 3 above)
Comment (sksandani) [reply 02_bubble/sksandani/main.go]: Refactored the code make copy of input

File: 02_bubble/sksandani/main.go (hunk 4 above)
Comment (sksandani) [reply 02_bubble/sksandani/main.go]: Refactored the code make copy of input

File: 02_bubble/sksandani/main_test.go (hunk 5 above)
Comment (sksandani) [reply 02_bubble/sksandani/main_test.go]: Updated the message

**Comment by juliaogris** [comment]
Click `Update branch` and locally `git pull origin lab2` before adding new commits.

File: 02_bubble/sksandani/main_test.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	actual := buf.String()
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+
+func TestBubbleSort(t *testing.T) {
+	// Given
+	r := require.New(t)
+
+	// When
+	i := bubble([]int{3, 2, 1, 5}) // Then
```
Comment (juliaogris) [inline 02_bubble/sksandani/main_test.go:30]: `actual := ...`

**Review by juliaogris — COMMENTED** [review]
Pretty close, update branch as described in previous comment.
Please add test to prove that input is not modified.
Add test for corner cases: empty slice, one element slice, slice with duplicated elements.
Consider using table tests.

**Comment by juliaogris** [comment]
Click `Update branch` and locally `git pull origin lab2` before adding new commits.
